### ISSUE TITLE
line breaks should be escaped in CSV exports

### ISSUE THREAD

**Issue by bernisys** [Cacti/cacti#5811]
# Describe the bug

Line breaks in the exported data fields will cripple the lines in the CSV output.

## To Reproduce

go to the devices list
export it as CSV
check the CSV output for lines not properly formed

If in this case the system-info OID contains line breaks, these will appear also unmodified in the output, crippling the line so that it cannot be imported properly into other applications. Following lines will only contain the rest of the text, until the end is reached and a new export-line is created.

## Expected behavior

Line breaks should be escaped or the output should be formatted in a way that a parser can clearly detect field borders. This is usually done by enclosing each field in quotes.

**Comment by TheWitness** [comment]
Which UI?

**Comment by bernisys** [comment]
Happening in both 1.2.18 (our prod, soon to be 27) and also 1.2.27  (our test/dev) - just exported the device list there and see the same intermediate line breaks appearing:

![image](https://github.com/user-attachments/assets/4a2664cc-176d-419a-9b4f-a6e21f495025)

p.S:
I added blank lines between the data-sets to let them appear more clearly.
You can see, if the System OID contains breaks, they are not escaped in the CSV.

**Comment by TheWitness** [comment]
Let me check that.  I think we fixed for Notes by removing the line breaks, but I did not check the SNMP sysDescr.

**Comment by TheWitness** [comment]
Yea, it's fixed in 1.2.28.  Replace the function in host.php with the following:

```php
function host_export() {
    host_validate_vars();

    $hosts = get_device_records($total_rows, 9999999);

    $stdout = fopen('php://output', 'w');

    header('Content-type: application/excel');
    header('Content-Disposition: attachment; filename=cacti-devices-' . time() . '.csv');

    if (cacti_sizeof($hosts)) {
        $columns = array_keys($hosts[0]);

        fputcsv($stdout, $columns);

        foreach($hosts as $h) {
            foreach(array_keys($h) as $hc) {
                if ($h[$hc] != '' && (strpos($h[$hc], "\n") !== false || strpos($h[$hc], "\r") !== false)) {
                    $h[$hc] = str_replace(array("\n", "\r"), ' ', $h[$hc]);
                }
            }

            fputcsv($stdout, $h);
        }
    }

    fclose($stdout);
}
```
